http/agent.go

```diff
@@ -308,3 +319,221 @@ func (a *Agent) PostToWriter(w io.Writer, url string, postData []byte) error {
+		if resps[i] == nil {
+			continue
+		}
+		resps[i].Body.Close()
+	}
+}
+
+// PostToWriterGroup behaves just as PostToWriter() but takes a group of URLs
+// and performs the requests in parallel. The number of simultaneous requests
+// is controlled by options.MaxParallel.
+//
+// The list of URLs and postData byte arrays are expected to be of equal length.
+// If postData has less elements than the url list, those urls without a corresponding
+// postData array will return an error.
+//
+// If the w writers slice contains a single writer, all the responses will be
+// written to the single writer. If the writers array contains more than one
+// io.Writer, each request will be written to its corresponding writer unless it
+// is missing, in that case the request will return an an error. The requests are
+// guaranteed to go into the writer in order.
+func (a *Agent) PostToWriterGroup(w []io.Writer, urls []string, postData [][]byte) []error {
+	//nolint: bodyclose // Next line closes them
+	resps, errs := a.PostRequestGroup(urls, postData)
+	defer closeHTTPResponseGroup(resps)
+
+	for i, r := range resps {
+		if r == nil {
+			continue
+		}
+
+		var err error
+		if len(w) == 1 {
+			err = a.readResponse(r, w[0])
+		} else {
+			if i >= len(w) {
+				err = fmt.Errorf("request %d has no writer defined", i)
+			} else {
+				err = a.readResponse(r, w[i])
+			}
+		}
+		if err != nil {
+			errs[i] = fmt.Errorf("writing group response #%d: %w", i, err)
+			continue
+		}
+	}
+	return errs
+}
+
+// GetGroup behaves just as Get() but takes a group of URLs and performs
+// the requests in parallel. The number of simultaneous requests is controlled by
+// options.MaxParallel.
+func (a *Agent) GetGroup(urls []string) ([][]byte, []error) {
+	//nolint: bodyclose // Next line closes them
+	resps, errs := a.GetRequestGroup(urls)
+	defer closeHTTPResponseGroup(resps)
+
+	c := make([][]byte, len(urls))
+	for i, r := range resps {
+		if r != nil {
+			d, err := a.readResponseToByteArray(r)
+			if err != nil {
+				errs[i] = fmt.Errorf("reading group response #%d: %w", i, err)
+				continue
+			}
+			c[i] = d
+		}
+	}
+	return c, errs
+}
+
+// GetToWriterGroup behaves just as GetToWriter() but takes a group of URLs
+// and performs the requests in parallel. The number of simultaneous requests
+// is controlled by options.MaxParallel.
+//
+// If the w writers slice contains a single writer, all the responses will be
+// written to the single writer. If the writers array contains more than one
+// io.Writer, each request will be written to its corresponding writer unless it
+// is missing in which case the request will return an an error. The requests are
+// guaranteed to go into the writer in order.
+func (a *Agent) GetToWriterGroup(w []io.Writer, urls []string) []error {
+	//nolint: bodyclose
+	resps, errs := a.GetRequestGroup(urls)
+	defer func() {
+		for i := range resps {
+			if resps[i] == nil {
+				continue
+			}
+			resps[i].Body.Close()
+		}
+	}()
```

Comment:
Why we don't use `closeHTTPResponseGroup` here?

Reply:
Ah good catch, I forgot to remove this inline func. Done.